### PR TITLE
* NEW [broker_test] add broker test in cmd way.

### PR DESCRIPTION
**Pull request by Hermann0222**

### CODE REVIEW

File: nanomq/apps/broker_test.c [hunk 1]
```diff
@@ -1,23 +1,90 @@
...
+#include <unistd.h>
 
 #include "include/broker.h"
 
-int main()
+#define READ 0
+#define WRITE 1
+
+pid_t
+popen2(const char *command, int *infp, int *outfp)
```
Comment (wanghaEMQ) [inline nanomq/apps/broker_test.c:13]: The name of the function is conflict with what it actually does

Comment (wanghaEMQ) [reply nanomq/apps/broker_test.c:13]: popen_for_sub()?

File: nanomq/apps/broker_test.c [hunk 2]
```diff
@@ -1,23 +1,90 @@
...
+	assert(strncmp(buf, "message", 7) == 0);
 
-	//  rv = broker_reload(0, NULL);
-	//  assert(rv == 0);
+	kill(pid_sub, SIGKILL);
+	pclose(p_pub);
 
-	rv = broker_dflt(0, NULL);
-	assert(rv == 0);
+	assert(broker_dflt(0, NULL) == 0);
```
Comment (wanghaEMQ) [inline nanomq/apps/broker_test.c:87]: Did you forget to close the infd or outfd?

File: nanomq/apps/broker.c [hunk 3]
```diff
@@ -1081,7 +1081,7 @@ broker(conf *nanomq_conf)
 	printf("NanoMQ Broker is started successfully!\n");
 
 #if defined(ENABLE_NANOMQ_TESTS)
-	bool is_testing = true;
+	bool is_testing = false;
 #else
 	bool is_testing = false;
 #endif
```
Comment (wanghaEMQ) [inline nanomq/apps/broker.c:1087]: Both are false? Are you sure? Delete behaviors related to this variable if yes

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]

File: nanomq/apps/broker.c (hunk 3 above)
Comment (Hermann0222) [reply nanomq/apps/broker.c]: just fix the compatibility issue with the previous codes.

File: nanomq/apps/broker_test.c [hunk 4]
```diff
@@ -1,23 +1,90 @@
...
 #include <assert.h>
+#include <pthread.h>
+#include <signal.h>
+#include <unistd.h>
 
 #include "include/broker.h"
 
-int main()
+#define READ 0
+#define WRITE 1
```
[the remark this answers is not in the document]
Comment (Hermann0222) [reply nanomq/apps/broker_test.c]: I will use  STDIN_FILENO and STDOUT_FILENO instead.

**Pull request merged**